src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioSolutionSizeTracker.cs

```diff
@@ -34,18 +37,10 @@ public SolutionSizeTracker()
         /// lazy and very cheap on answering that question.
         /// </summary>
         public long GetSolutionSize(Workspace workspace, SolutionId solutionId)
-        {
-            var service = workspace.Services.GetService<IPersistentStorageLocationService>();
-            return service.IsSupported(workspace)
-                ? _tracker.GetSolutionSize(solutionId)
-                : -1;
-        }
+            => workspace is VisualStudioWorkspace ? _tracker.GetSolutionSize(solutionId) : -1;
```

Comment:
Can't we just check the workspace kind is host?

Reply:
We could.  I'm just not certain waht that would cahnge.  The previous logic had these parts:

1. The `storage service`.  it would always run in oop and only run in proc if the `location service` was supported and the `soulution size service` reported a large enough solution size.
2. the `solution size service`. it only computed the size if the `location service` was supported.
3. the `location service`. it only returned that it was supported for vs workspaces.  i.e.: https://github.com/dotnet/roslyn/pull/39655/files/8ed4b05743e610af3b19a6743942b82ddcf7bb5a#diff-72ffd49f1f484d53fc7e000c8e2056c9L42

So it's unclear to me what the difference in behavior here is between what we had then, and what we have now here.